### PR TITLE
Fixes #25973 - add helpers for errata report

### PR DESCRIPTION
**Pull request by ares**
requires changes from https://github.com/theforeman/foreman/pull/6458, best tested with report template I'll submit in community-template and link from there

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

**Comment by ares** [comment]
[test katello] database does not exist...

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
Looks like rubocop had some failures.  Going to test this one out.

**Comment by ares** [comment]
Thanks, the best way to test is getting the report template using this from community templates repo. The foreman core part is already in.

**Comment by ares** [comment]
rubocop fixed, anything else? :-)

File: db/seeds.d/102-organizations.rb [hunk 1]
```diff
@@ -12,12 +19,12 @@
...
-end
+#if ENV['SEED_ORGANIZATION']
+#  if Setting['db_pending_seed']
+#    admin = User.where(:login => ENV['SEED_ADMIN_USER'].present? ? ENV['SEED_ADMIN_USER'] : 'admin').first
+#    if admin && admin.default_organization.nil?
+#      admin.default_organization = Organization.find_by(:name => ENV['SEED_ORGANIZATION'])
+#      admin.save!
+#    end
+#  end
+#end
```
Comment (jlsherrill) [inline db/seeds.d/102-organizations.rb]: I don't think you meant to include these changes :)

**Comment by ares** [comment]
oops, sorry, removed the addition :-)

**Comment by jlsherrill** [comment]
This works well, and the code here looks good.  A couple questions:

1) Should the errata report templates end up katello?  I see they seem to be getting synced to foreman right now.  This is more for my understanding.

2) the last reboot time seems to be using dates from ansible or puppet, however subscription-manager seems to include this date anyways (and sub-man is required for errata applicability).  So it seems like it would make a lot of sense to use that one.   It uses the proc::btime fact:

```
proc_stat.btime: 1550497541
```

I notice in the PR that it was mentioned that plugins should be able to extend this, but in the meantime could we add this sub-man fact to the list in core foreman?

**Comment by ares** [comment]
Thanks Justin. Answers:

1) all templates should primarilly live in community-templates repo. Since we added requirements to metadata, templates can now specify they require specific plugin and potentially even a version. So we sync them all to Foreman and if requirements are not met, we don't seed them.

2) I have missed this fact when I was looking. It totally makes sense to look at subscription manager fact too, I just don't know, if it should prefer it. How often are these facts updated? Puppet updates by default every 30 minutes. I'll open separate PR to Foreman and ping you for review if you don't mind.

**Comment by jlsherrill** [comment]
@ares every 4 hours by default.  I think it makes sense to use the puppet fact if its there, but if its not use the sub-man fact.

**Comment by jlsherrill** [comment]
Thanks @ares !

**Pull request merged**